nix/build.sh

```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/env bash
+
+set -e
+
+# cleanup for artefacts created during builfs
+function cleanup() {
+  trap - EXIT ERR INT QUIT
+
+  if [ -n "$nixResultPath" ]; then
+    echo "Deleting derivations from Nix store..."
+    releaseDrv=$(nix-instantiate --quiet $nixOpts)
```

Comment:
It's going to be challenging to make the deletion logic generic, especially because I think nix-instantiate will kick off a build if necessary. Since in this case we're calling it with the same `$nixOpts` it's fine, but we could rename `releaseDrv` to `buildDrv`.

Reply:
Yeah, I haven't thought about that part that much, I thought you could help me of thinking of a general logic for cleaning stuff we don't want.